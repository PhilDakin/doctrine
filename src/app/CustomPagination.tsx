import { Box, Paper, Button, Typography } from "@mui/material";
import usePagination from "@mui/material/usePagination/usePagination";
import { Dispatch, SetStateAction } from "react";
import ChevronRightIcon from "@mui/icons-material/ChevronRight";
import ChevronLeftIcon from "@mui/icons-material/ChevronLeft";

type CustomPaginationProps = {
  setPage: Dispatch<SetStateAction<number>>;
};

export default function CustomPagination({ setPage }: CustomPaginationProps) {
  const { items } = usePagination({
    count: 3,
    onChange: (_, p) => setPage(p),
  });

  function getLabel(page: number | null) {
    if (!page) {
      console.error("Trying to get page number for null page.");
      return;
    }

    if (page === 1) {
      return "Original Text";
    } else {
      return "Summary #" + (page - 1);
    }
  }

  return (
    <Box sx={{ width: "70%", display: "flex", justifyContent: "center" }}>
      {items.map(({ page, type, selected, ...item }, index) => {
        if (type === "start-ellipsis" || type === "end-ellipsis") {
          // TODO (pdakin): Render something for large page counts.
          return null;
        }

        let content =
          type === "page" ? (
            getLabel(page)
          ) : type === "previous" ? (
            <ChevronLeftIcon />
          ) : (
            <ChevronRightIcon />
          );
        let width = type === "page" ? "100%" : "5%";
        let elevation = type === "page" ? 3 : 0;
        return (
          <Paper
            key={index}
            sx={{ marginX: 0.45 }}
            square
            elevation={elevation}
          >
            <Button
              sx={{ whiteSpace: "nowrap", width: width }}
              type="button"
              {...item}
            >
              <Typography
                sx={{ fontWeight: selected ? "bold" : undefined }}
                color="text.primary"
              >
                {content}
              </Typography>
            </Button>
          </Paper>
        );
      })}
    </Box>
  );
}
