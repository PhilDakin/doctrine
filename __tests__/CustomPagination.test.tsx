import renderer from "react-test-renderer";
import CustomPagination from "@/app/CustomPagination";

it("CustomPagination snapshot match", () => {
  const component = renderer.create(
    <CustomPagination setPage={jest.fn()}></CustomPagination>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
