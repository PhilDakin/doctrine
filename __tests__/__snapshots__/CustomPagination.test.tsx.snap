// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomPagination snapshot match 1`] = `
<div
  className="MuiBox-root css-x4r31f"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 css-69z8z6-MuiPaper-root"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1ht921-MuiButtonBase-root-MuiButton-root"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ChevronLeftIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
          />
        </svg>
      </p>
    </button>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation3 css-1m8jwf8-MuiPaper-root"
  >
    <button
      aria-current="true"
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1w7x8rb-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-3vksh9-MuiTypography-root"
      >
        Original Text
      </p>
    </button>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation3 css-1m8jwf8-MuiPaper-root"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1w7x8rb-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
      >
        Summary #1
      </p>
    </button>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation3 css-1m8jwf8-MuiPaper-root"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1w7x8rb-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
      >
        Summary #2
      </p>
    </button>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 css-69z8z6-MuiPaper-root"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1ht921-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ChevronRightIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
      </p>
    </button>
  </div>
</div>
`;
